### DISCUSSION TITLE
Mypy and SQLAlchemy 2.0

### DISCUSSION THREAD

**Discussion by fcs-ts** [sqlalchemy/sqlalchemy#9853] [Usage Questions]
I am trying to update sqlalchemy 1.46.x to 2.0.15 and I am dealing with typing (mypy) issues. I've dealt with most of them, but I have a few remaining related to

-    dialect specific calls: we use postgres insert to leverage the on_conflict functionality
-    sql functions: sum, coalesce, filter`

Do anyone here have some experience or some pointers on how to deal with those?

**Comment by CaselIT** [comment]
Hi,

as mentioned in the chat, on_conflict is likely not typed, while the common functions should be typed.

To have more information you need to provide specifics on the errors

**Comment by fcs-ts** [comment]
Thanks for your quick reply. Let's start with `sum`, with a kind of minimal example on how it is being used:
```python
from sqlalchemy import BigInteger, Column, MetaData, Table, select, text
from sqlalchemy.sql.functions import sum

import app.db

metadata_obj = MetaData()
table = Table(
    "table_name",
    metadata_obj,
    Column("quantity", BigInteger, nullable=False, server_default=text("0")),
)


async def do_query() -> int:
    query = select(sum(table.c.quantity))
    async with app.db.async_engine.begin() as conn:
        rows = (await conn.execute(query)).first()
        assert rows
    return int(rows)
````

That produces:
````
sql.py:15: error: Need type annotation for "query"  [var-annotated]
sql.py:15: error: Call to untyped function "sum" in typed context  [no-untyped-call]
Found 2 errors in 1 file (checked 1 source file)
````

**Reply by CaselIT** [reply]
indeed it's not. I guess I miss-remembered. It's listed here and all https://github.com/sqlalchemy/sqlalchemy/issues/6810
